### PR TITLE
feat: added redeem reward flow

### PR DESCRIPTION
**Pull request by JasperVercammen**
redeemreward

### CODE REVIEW

File: src/redeemedReward/RedeemedReward.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import { ScrollView } from 'react-native';
+
+import * as Styled from './style';
+
+const RedeemedReward = () => {
+  return (
+    <ScrollView>
+      <Styled.Label>Succesvol omgeruild, dit scherm wordt snel opgepakt om er mooier uit te zien!</Styled.Label>
```
Comment (JasperVercammen) [inline src/redeemedReward/RedeemedReward.tsx:9]: UIteraard temp. Volgende PR pakt dit scherm op

Comment (samvanhoeyicapps) [reply src/redeemedReward/RedeemedReward.tsx:9]: 👍

File: src/shopDetail/ShopDetail.tsx [hunk 2]
```diff
@@ -19,52 +22,74 @@ export const ShopDetail = ({ route }: TProps) => {
   const { id, reward: fallbackReward } = route.params;
   const { data } = useGetReward({ id });
   const reward = data || fallbackReward;
+  const { data: redeemStatus, isLoading: isRedeemStatusLoading, error: redeemError } = useGetRedeemStatus({ id: reward.id });
+  const [isRedeemModalConfirmationOpen, toggleRedeemModalConfirmationOpen] = useToggle(false);
```
Comment (samvanhoeyicapps) [inline src/shopDetail/ShopDetail.tsx:26]: vind dit geen toggle 🙈

File: src/shopDetail/_models/redeemStatus.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export type TRedeemReason =
```
Comment (samvanhoeyicapps) [inline src/shopDetail/_models/redeemStatus.ts:1]: mooie comments 👏

**Review by samvanhoeyicapps — APPROVED** [review]

File: src/shopDetail/_components/redeemModal/RedeemModal.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  rewardId: string;
+  title: string;
+  toggleIsVisible: () => void;
+};
+
+const RedeemModal: FC<TRedeemModalProps> = ({ isVisible, toggleIsVisible, rewardId, points, title }) => {
+  const { navigate } = useNavigation<TRootStackNavigationProp>();
+
+  const { mutate: redeemReward, isLoading, error } = useRedeemReward({
+    onSuccess: () => {
```
[the remark this answers is not in the document]
Comment (JasperVercammen) [reply src/shopDetail/_components/redeemModal/RedeemModal.tsx:24]: "Before you jump on the bandwagon, please note that we're talking only about the callbacks of useQuery, NOT about the useMutation callbacks. Once I explained this and the drawbacks about those callbacks, most people tend to agree with that decision."
-> dit is mutation

**Pull request merged**